examples/ct/handlers_test.go

```diff
@@ -88,25 +88,36 @@ LPd2G+er1/w5wxpM/hvZbWc0yoLyLd5aDIu73YJde28+dhKtjbMAp+IRaYhgIyYi
 hMOqXSGR79oQv5I103s6KjQNWUGblKSFZvP6w82LU9Wk6YJw6tKXsHIQ+c5KITix
 iBEUO5P6TnqH3TfhOF8sKQg=`
 
-const caAndIntermediateCertsPEM string = "-----BEGIN CERTIFICATE-----\n" + caCertB64 + "\n-----END CERTIFICATE-----\n" +
-	"\n-----BEGIN CERTIFICATE-----\n" + intermediateCertB64 + "\n-----END CERTIFICATE-----\n"
+const caAndIntermediateCertsPEM string = "-----BEGIN CERTIFICATE-----\n" +
+	caCertB64 +
+	"\n-----END CERTIFICATE-----\n" +
+	"\n-----BEGIN CERTIFICATE-----\n" +
+	intermediateCertB64 +
+	"\n-----END CERTIFICATE-----\n"
 
 type handlerTestInfo struct {
 	mockCtrl *gomock.Controller
-	km       *crypto.MockKeyManager
+	km       *crypto.MockPrivateKeyManager
 	roots    *PEMCertPool
 	client   *mockclient.MockTrillianLogClient
 	c        LogContext
 }
 
 // setupTest creates mock objects and contexts.  Caller should invoke info.mockCtrl.Finish().
 func setupTest(t *testing.T, pemRoots []string) handlerTestInfo {
+
 	info := handlerTestInfo{}
 	info.mockCtrl = gomock.NewController(t)
-	info.km = crypto.NewMockKeyManager(info.mockCtrl)
-	info.km.EXPECT().GetRawPublicKey().AnyTimes().Return([]byte("key"), nil)
+
+	info.km = crypto.NewMockPrivateKeyManager(info.mockCtrl)
+	pubkey, err := crypto.PublicKeyFromPEM(ctTesttubePublicKey)
+	if err != nil {
+		panic(err)
+	}
+	info.km.EXPECT().PublicKey().AnyTimes().Return(pubkey)
```

Comment:
I've noticed that some of our mock expectations are completely unnecessary. By saying `AnyTimes()`, the method doesn't ever have to be called during the test. I'm not sure whether this is a good or bad thing. Unnecessary code is generally bad, but I suppose this makes the tests more resilient to changes. Fundamentally, we're not using these mocks as mocks a lot of the time, but as stubs instead, which is probably where this oddity comes from.